### PR TITLE
Feature/auto suggestion form

### PR DESCRIPTION
**Pull request by silksil**
- make auto suggestion work with react hook form
- implement it on the search page
- link to homepage when clicking on logo

### CODE REVIEW

**Review by bobetbat — COMMENTED** [review]
https://nominatim.openstreetmap.org/search?format=json&q=${location} . this is api endpoint that returns us list of results based on search, by default it returns array response  with autocomplete( and now it is set to pick first element bc its likely the thing user is looking for. So my suggestion is, to trigger this endpoint and return autocomplete results, so when user will choose it, it will finally go to our accommodation and offer endpoint with data in auto complete

File: src/components/hook-form/RHFAutocomplete.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  width?: string | number;
+};
+
+type Props = IProps & TextFieldProps;
+
+export default function RHFTAutocomplete({
+  name,
+  options,
+  freeSolo = true,
+  width = '300px',
```
Comment (bobetbat) [inline src/components/hook-form/RHFAutocomplete.tsx:17]: I would use mui numbers, and avoid hardcoded px

File: src/containers/search/SearchForm.tsx [hunk 2]
```diff
@@ -220,10 +222,12 @@ export const SearchForm: React.FC = () => {
               />
             }
           >
-            <RHFTextField
+            <RHFTAutocomplete
               variant="standard"
               placeholder="Where are you going?"
               name="location"
+              options={autocompleteData}
+              width="230px"
```
Comment (bobetbat) [inline src/containers/search/SearchForm.tsx:230]: I would use mui numbers, and avoid hardcoded px (2)

File: src/containers/search/SearchForm.tsx [hunk 3]
```diff
@@ -247,7 +251,7 @@ export const SearchForm: React.FC = () => {
                 size={buttonSize}
                 variant="text"
                 sx={{
-                  minWidth: '210px',
+                  minWidth: '230px',
```
Comment (bobetbat) [inline src/containers/search/SearchForm.tsx:254]: I would use mui numbers, and avoid hardcoded px (3)

Comment (silksil) [reply src/containers/search/SearchForm.tsx:254]: MUi numbers are great for spacing between elements. This is really styling a specific elmeent.  Therefore it also is used as `spacing(1)`. 

See: https://mui.com/material-ui/customization/spacing/

File: src/components/hook-form/RHFAutocomplete.tsx (hunk 1 above)
Comment (silksil) [reply src/components/hook-form/RHFAutocomplete.tsx:17]: see comment below

**Pull request merged**